flycheck.el

```diff
@@ -9329,6 +9329,8 @@ This syntax checker requires Rust 1.15 or newer.  See URL
   :enabled (lambda ()
              (-when-let (file (buffer-file-name))
                (locate-dominating-file file "Cargo.toml")))
+  :working-directory (lambda (_)
+                       (locate-dominating-file (buffer-file-name) "Cargo.toml"))
```

Comment:
Can we assume that we have a buffer-file-name when `:working-directory` is set? If so, should we document this? (presumably in `flycheck-define-generic-checker` we'd say that the function is only run if the `:predicate` and `:enabled` functions ran.)

Reply:
I don't think we can assume that, no.  I just tested it, and it can definitely fail if the buffer has no file name.

In fact, looking at `flycheck-define-generic-checker`, I see that `working-directory` is used as the default directory for the `:enabled` and `:predicate` predicates, so `:working-directory` must guard `buffer-file-name`, otherwise the `enabled` check may fail.

Reply:
OK, yay for code review then ^^